compiler/p/codegen/OMRLoadStoreHandler.hpp

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_POWER_LOADSTOREHANDLER_INCL
+#define OMR_POWER_LOADSTOREHANDLER_INCL
+
+#include "codegen/InstOpCode.hpp"
+#include "codegen/MemoryReference.hpp"
+#include "codegen/Register.hpp"
+#include "il/Node.hpp"
+#include "infra/Annotations.hpp"
+
+namespace OMR
+{
+
+namespace Power
+{
+class NodeMemoryReference
+    {
+    TR::MemoryReference *_memRef;
+
+    public:
+    NodeMemoryReference(TR::MemoryReference *memRef) : _memRef(memRef) {}
```

Comment:
I don't know if this is feasible, perhaps you may be able to answer this. Is it possible to use RAII here to carry out the cleanup for reference counts? I see that in most (all?) places this is now used we are having to call `ref.decReferenceCounts(cg)` right before this object goes out of scope.

Reply:
I did think about this. While it may seem like a good idea at first, decrementing the refcounts isn't really a cleanup task we want to be running if an exception is thrown to abort the compilation. The logic of decrementing a refcount in the codegen is quite complex and doing that would risk reintroducing a variant of #4846.